mesh.py

```diff
@@ -28,7 +32,50 @@ def __init__(self, mesh_cells, domain_upper, mat_map):
     self._y_node = mesh_cells + 1
     self._n_node = self._x_node * self._y_node
     self._cell_length = self._mesh_params['cell_length']
+
```

Comment:
Would you change the indentation to be 4 per block of code. I know this is annoying, but it would be easier to see since Python does not allow using {} for different scopes.

Reply:
Do you mean to put four newlines between the different functions? Like
```
self._cell_length = self._mesh_params['cell_length']



def soln_plot(self, solution, plot = True): # pragma: no cover
```

Reply:
No, I meant in some part you have
`
def func():
    # do something after 4 indentations, which is good
`
Some part
`
def func():
  # do something after 2 indentations, which is not consistent with previous 4 indentations
`